cloaked.js

```diff
@@ -1,93 +1,122 @@
+		// Creates an observer, and tells it what to do when it successfully observes mutations
+		// Necessary for Facebooks "neverending" scrolling
+		var observer = new MutationObserver(function (mutations, observer) {
+			// fired when a mutation occurs
+			count++;
+			
+			// For user news feed
+			if (feed && (count % FILTER_DELAY === 0)) {
+				blockFacebookSpoilers("[id^=hyperfeed_story_id_]");
+			}
+
+			// For page feeds
+			if (page && (count % FILTER_DELAY === 0)) {
+				blockFacebookSpoilers("[class^=_4-u2]");
+			}
+
+			// Will potentially need to put in a check for _401d feed and _307z posts
+		});
+		
+		// Setup the observers to observe specific targets
+		if (feed) {
+			setupObserver(feed, observer);
+		}
+
+		if (page) {
+			setupObserver(page, observer);
+		}
+			
+	}
+
+	// This function applies the censor filters to the page based on the items in the spoiler list.
+	var searchForSpoilers = function() {
+		if (SPOILER_LIST == null) {
+			return null;
+		}
+		var searchString = '';
+		SPOILER_LIST.forEach(function (item) {
+			
+			searchString = searchString + [
+								"p:icontains('" + item + "')", 
+								"h1:icontains('" + item + "')", 
+								"h2:icontains('" + item + "')", 
+								"li:icontains('" + item + "')", 
+								"span:icontains('" + item + "')", 
+								"img[src*='" + item + "']", 
+								"source[src*='" + item + "'], "
+							].join(', ');
+		});
+		searchString = searchString.substring(0, searchString.length - 2);
+
+		// Blur the spoiler's parent (excluding body+head) because it's important to block the spoiler's surrounding terms.
+		$(searchString).parent(":not('body')", ":not('head')").css('-webkit-filter', 'blur(5px)').click(function() {
+			$(this).css('-webkit-filter', '');
+		});
+	}
+
+	// searchForSpoilers and blockFacebookSpoilers still need to be refactored into one function
+	var blockFacebookSpoilers = function(blockElement) {
+		if (SPOILER_LIST == null) {
+			return null;
+		}
+		var searchString = '';
+		SPOILER_LIST.forEach(function (item) {
+			
+			searchString = searchString + "p:icontains('" + item + "'), span:icontains('" + item + "'), ";
+		});
+		searchString = searchString.substring(0, searchString.length - 2);
+		$(searchString).parents(blockElement).css('-webkit-filter', 'blur(5px)');
+	}
+
+	// This function checks if the title includes any of the terms, if it does, then it will block all the images and videos on the page - just incase.
+	checkTitle = function() {
+		var title = $('title').text().toLowerCase();
+		for (var i = 0; i < SPOILER_LIST.length; i++) {
+			if (title.includes(SPOILER_LIST[i])) {
+				$('img, source').parent(":not('body')", ":not('head')").css('-webkit-filter', 'blur(5px)');
+				break;
+			}
+		}
+	}
+
+	// Case insensitive jquery contains
+	jQuery.expr[':'].icontains = function(element, index, match) {
+		return jQuery(element).text().toUpperCase()
+		.indexOf(match[3].toUpperCase()) >= 0;
+	};
+
+	// Run script
+	$(function () {
```

Comment:
Is this a separate thing from the `(function ($) {` near the top of the page?

Reply:
Yes, it's another self executing function